kernel/src/hil/i2c.rs

```diff
@@ -56,6 +59,9 @@ pub trait I2CMaster {
 
 /// Interface for an I2C Slave hardware driver.
 pub trait I2CSlave {
+    fn set_slave_client(&self, _slave_client: Option<&'static dyn I2CHwSlaveClient>) {
```

Comment:
This should be here for exactly the reason you indicated, but we should remove the default implementation and instead implement them for each chip that needs them.

Reply:
I'll add them to the chips and resubmit this.

Reply:
I added the implementations to cc26x2, saml4 and nr52. I don't have the chips, so I cannot test it.

I modified the set_client functions so that they are consistent. Initially the set_master function received a I2CHwMasterClient and the set_slave_client received an Option<I2CHwMasterClient>. I modified both to receive a I2CHwMaster. Was there a reason for them being different? SPI is the same.